packages/typescript-estree/src/convert.ts

```diff
@@ -1167,12 +1167,30 @@ export class Converter {
 
       case SyntaxKind.PropertyDeclaration: {
         const isAbstract = hasModifier(SyntaxKind.AbstractKeyword, node);
+        const isAccessor = hasModifier(SyntaxKind.AccessorKeyword, node);
+
+        // eslint-disable-next-line @typescript-eslint/explicit-function-return-type -- TODO - add ignore IIFE option
```

Comment:
> `TODO - add ignore IIFE option`

What does this mean?

Reply:
right now EFRT will error on IIFE's

IIFEs really shouldn't be covered by the rule because there's no "contract" that needs to be enforced - there's no external callers or anything of the sort.

Additionally most IIFE cases you are just using an IIFE to wrap-up some logic for a specific variable to be initialised with a value. By forcing you to annotate an IIFE's return essentially you're forcing someone to annotate the variable declaration explicitly (which isn't what the rule is about!)

So I want to add an option to ignore IIFEs entirely in the rule.
Filed #6115